docs/source/contributing.rst

```diff
@@ -0,0 +1,187 @@
+
+Adding Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you are adding an new test file (e.g. for a new processor), then this needs to be added to the Test Suite in ``start_tests.py``.
+
+Provided Base Classes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Processor Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All processor tests inherit from the ``ProcessorTest`` class. Processors should create a ``Mock()`` object, which will contain the bare minimum for the processor to be run (it's jinja template's and properties loaded from ``processor-info.json``), i.e. there is no reason for it to know about properties of the other processors.
+
+A test method will typically follow the same sequence of steps:
+
+	1. Retrieve the test string (there is a ``read_test_file()`` method provided by the ``ProcessorTest`` class)
+	2. Split into blocks (there is a ``to_blocks()`` method provided by the ``ProcessorTest`` class)
+	3. Use an ``Assert`` to confirm there are(not) matches to the regex
+	4. Convert the test string using the ``kordac_extension`` (provided by the ``SetUp()`` method in ``ProcessorTest``)
+	5. Load the expected converted result
+	6. Check the converted result is the same as the expected result
+
+
+Testing Assets
+***************************************
+<explain markdown and html file >
+
+
+The test markdown file and expected html file should be placed in ``kordac/tests/assets/<processor-name>/``. The expected file should have the same name as the corresponding test file, with ``expected`` appended to the file name.
+
+For example:
+
+.. code-block:: none
+	
+	kordac/tests/assets/boxed-text/no_boxed_text.md
+	kordac/tests/assets/boxed-text/no_boxed_text_expected.html
+
+
+
+Creating a release
+=======================================
+
+This is our current process for creating and publishing a Kordac release. This
+can only be performed by repository administrators
+
+1. `Create a release branch <http://nvie.com/posts/a-successful-git-branching-model/#creating-a-release-branch>`_. Checkout to this branch.
+2. Update the version number [1]_ within ``kordac/__init__.py``.
+3. Check test suite for errors, and fix any issues that arise, or `log an issue <https://github.com/uccser/cs-field-guide/issues/new>`_.
+4. Detail the changes in ``docs/source/changelog.rst``.
+5. `Complete the release branch <http://nvie.com/posts/a-successful-git-branching-model/#finishing-a-release-branch>`_. Be sure to tag the release with the version number for creating the release on GitHub.
+6. Create the release on `GitHub <https://github.com/uccser/kordac/releases/>`_ on the tagged commit.
+7. Upload a new version of Kordac to PyPI.
+
+.. [1] We follow `Semantic Versioning <http://semver.org/>`_ for our numbering system. The number is used by ``setup.py`` to tell PyPI which version is being uploaded or ``pip`` which version is installed, and also used during the documentation build to number the version of Kordac it was built from.
+
+
+Notes
```

Comment:
- https://help.github.com/articles/setting-guidelines-for-repository-contributors/
- https://help.github.com/articles/creating-an-issue-template-for-your-repository/
- https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/

We currently use all these for the Computer Science Field Guide.